website/pages/blog/building-cloudquery.md

```diff
@@ -0,0 +1,102 @@
+- **Scaling Destinations:** Unlike sources, destinations are finite but the right architecture should ensure source and destinations are decoupled correctly so new destinations will work out-of-the-box and won't require any changes in sources (otherwise development work will grow exponentially).
+- **Performance:** Running ELT workload is compute intensive and can get expensive depending on the number of APIs to extract from. Having high performance and concurrent architecture is key to both making cost as low as possible and bringing information as fast as possible from sources to destinations.
+
+# Technical Deep Dive
+
+This section will discuss the most important decisions and solutions we landed on, organized by the challenges listed above.
+
+## API Coverage
+
+### Open Source
+
+There is no vendor that can support all integrations in-house thus the best and only way in our opinion is to go the open-source way and give the ability for others to contribute to our official plugins and develop community plugins without being blocked by a single vendor. Good recent examples that dealt with the “infinite APIs integrations” problem and took the open source route successfully are Terraform and Pulumi.
+
+### Pluggable Sources
+
+Pluggable sources are key to scale API coverage both from development and usability perspective and solve the following challenges:
+
+- **Binary Size:** Users download only the plugins they want to use to ensure good performance and user experience without a binary that grows infinitely and will eventually cause not only binary size issues but library clashes from different services.
+- **Versioning:** Given plugins are developed as standalone gRPC binaries users can use different versions of different plugins depending on their needs.
+- **Independent development:** Developers can develop their own plugins in their own repositories without being blocked by a vendor (us in this case) to review PRs.
+
+Our [pluggable system](https://www.cloudquery.io/docs/developers/architecture) is based on gRPC to ensure our plugins can be cross-platform, independent  and performant.
+
+### SDK for Source Plugins
+
+Writing ELT code involves a lot of boilerplate code, testing code, tricky performance and parallelization code. We’ve built a [Go SDK](https://github.com/cloudquery/plugin-sdk/) that enables both us and other developers to focus only on the E(Extract) code while CloudQuery SDK takes care of parallelization, transformation, loading and testing.
+
+### Code Generation
+
+Big challenge around scaling support for services and APIs support for highly different APIs and keeping up the code with the server APIs and schema. Before we dive into how we leveraged code-generation to generate CloudQuery plugins, let’s take a quick look at how client library generation for multiple languages worked in those days.
```

Comment:
I don't believe I understand the first sentence here.

Reply:
simplified